app/controllers/downloads_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+class DownloadsController < ApplicationController
+  include Hydra::Controller::DownloadBehavior
+
+  def show
+    send_content
+  end
+
+  class FileWithMetadata < Dry::Struct
+    delegate :size, :read, :stream, to: :file
+    attribute :id, Valkyrie::Types::Any
+    attribute :file, Valkyrie::Types::Any
+    attribute :mime_type, Valkyrie::Types::Any
+    attribute :original_name, Valkyrie::Types::Any
+  end
+
+  def load_file
+    binary_file = storage_adapter.find_by(id: file_identifier)
+    FileWithMetadata.new(id: file_identifier, file: binary_file, mime_type: asset.mime_type, original_name: asset.original_filename)
+  end
+
+  # Customize the :download ability in your Ability class, or override this method
+  def authorize_download!
+    authorize! :download, asset
+  end
+
+  def prepare_file_headers
+    send_file_headers! content_options
+    response.headers['Content-Type'] = file.mime_type
+    response.headers['Content-Length'] ||= file.size.to_s
+    # Prevent Rack::ETag from calculating a digest over body
+    response.headers['Last-Modified'] = modified_date
+    self.content_type = file.mime_type
+  end
+
+  def modified_date
+    return unless asset.respond_to?(:modified_date)
+    asset.modified_date.utc.strftime("%a, %d %b %Y %T GMT")
```

Comment:
Copied/pasted from Hydra::Head. I'm inclined to leave it since it's purely frontend code that will be different in the Hyrax implementation anyways - and this override will probably go away as soon as #68 is closed.